ledger-core/vanilla/cachekit/core.go

```diff
@@ -163,6 +163,16 @@ func (p *Core) addToGen(index Index, addedEntry, overflow bool) GenNo {
 func (p *Core) useOrAdvanceGen(age Age, added bool) {
 	trimmedEntries, trimmedGens := false, false
 
+	if p.oldest == nil {
```

Comment:
well, it will panic with nil anyway ... here and in the other methods

also, I'd suggest to use switch {} for multiple checks - this is both go-style and easier to read

Reply:
I want to know if it's nil as soon as possible.